pkg/applicationdefinitions/application_catalog.go

```diff
@@ -0,0 +1,131 @@
+		b, err := io.ReadAll(file)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read ApplicationDefinition: %w", err)
+		}
+
+		appDef := &appskubermaticv1.ApplicationDefinition{}
+
+		err = yaml.Unmarshal(b, appDef)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse ApplicationDefinition: %w", err)
+		}
+
+		creators = append(creators, systemApplicationDefinitionReconcilerFactory(appDef, config))
+	}
+
+	return creators, nil
+}
+
+// systemApplicationDefinitionReconcilerFactory creates a reconciler that handles system ApplicationDefinitions.
+func systemApplicationDefinitionReconcilerFactory(
+	fileAppDef *appskubermaticv1.ApplicationDefinition,
+	config *kubermaticv1.KubermaticConfiguration,
+) kkpreconciling.NamedApplicationDefinitionReconcilerFactory {
+	return func() (string, kkpreconciling.ApplicationDefinitionReconciler) {
+		return fileAppDef.Name, func(clusterAppDef *appskubermaticv1.ApplicationDefinition) (*appskubermaticv1.ApplicationDefinition, error) {
+			l := fileAppDef.GetLabels()
+			if l == nil {
+				l = make(map[string]string)
+			}
+			l[appskubermaticv1.ApplicationManagedByLabel] = appskubermaticv1.ApplicationManagedByKKPValue
+			fileAppDef.SetLabels(l)
+
+			// Labels and annotations specified in the ApplicationDefinition installed on the cluster are merged with the ones specified in the ApplicationDefinition
+			// that is generated from the system applications.
+			kubernetes.EnsureLabels(clusterAppDef, fileAppDef.Labels)
+			kubernetes.EnsureAnnotations(clusterAppDef, fileAppDef.Annotations)
+
+			// State of the following fields in the cluster has a higher precedence than the one coming from the default application catalog.
+			if clusterAppDef.Spec.Enforced {
+				fileAppDef.Spec.Enforced = true
+			}
+
+			if clusterAppDef.Spec.Default {
+				fileAppDef.Spec.Default = true
+			}
+
+			if clusterAppDef.Spec.Selector.Datacenters != nil {
+				fileAppDef.Spec.Selector.Datacenters = clusterAppDef.Spec.Selector.Datacenters
+			}
+
+			// Update the application definition (fileAppDef) according to the KubermaticConfiguration.
+			// When KubermaticConfiguration includes HelmRegistryConfigFile, we need to update the application definition
+			// to include the Helm credentials provided by the user in the cluster.
+			updateApplicationDefinition(fileAppDef, config)
+
+			// Also, we need to ensure that the default values are set correctly. To do this:
+			// 1. Get the default values from the currently being reconciled application definition (clusterAppDef)
+			// 2. If it's empty, use `fileAppDef` default values as a source of truth - ensuring the values are never empty
+			// 3. If it's not empty, use the current values from the reconciled object, allowing users to override the default values
+			if clusterAppDef.Spec.DefaultValuesBlock != "" && clusterAppDef.Spec.DefaultValuesBlock != "{}" {
+				fileAppDef.Spec.DefaultValuesBlock = clusterAppDef.Spec.DefaultValuesBlock
+			}
```

Comment:
Sidenote: When you create a testing ticket for this please make sure you mention that these conditions to ensure that operator doesn't revert `DefaultValuesBlock` if a KKP admin modified it, is also tested/verified.